bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RoundRobinDistributionSchedule.java

```diff
@@ -138,4 +116,64 @@ public QuorumCoverageSet getCoverageSet() {
     public boolean hasEntry(long entryId, int bookieIndex) {
         return getWriteSet(entryId).contains(bookieIndex);
     }
+
+    private class AckSetImpl implements AckSet {
```

Comment:
My feeling is -- the `lastAddSyncedMap` should not be maintained per entry and the calculation of `LastAddSynced` should not be done per entry.

I think it should be maintained in `LedgerHandle` - each time a bookie advanced its `LastAddSynced`, it will recalculate a new `LastAddConfirmed` based on the overview of `LastAddSynced`, then it advances the pointer according to that.

Reply:
mmm, I think you suggestion is good.
I will try to move this logic into LedgerHandle and not per entry